BusBoard/BusBoard/TflApi.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using RestSharp;
+
+namespace BusBoard
+{
+    public class TflApi : IApiHandler
+    {
+        private static string _url = @"https://api.tfl.gov.uk";
+        private IRestClient _client = new RestClient(_url);
+        
+        public string BaseUrl {
+            get => _url;
+            set => _url = value;
+        }
+        
+        public IRestClient Client
+        {
+            get => _client;
+            set => _client = value;
+        }
+        
+        public T Execute<T>(RestRequest request) where T : new()
+        {
+            var response = _client.Execute<T>(request);
```

Comment:
This is a very general approach. I think we can just go with 

`client.Execute<List<IncomingBusPrediction>>(request).Data`

which will return a list of the bus predictions for you

Reply:
This means that you can get rid of the `GetIncomingBusPredictions` method